File: banchus-app/src/main/java/pe/nanamochi/banchus/protocol/HandleClientPacket.java
package pe.nanamochi.banchus.protocol;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.springframework.stereotype.Component;
import pe.nanamochi.banchus.core.Packets;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Component
public @interface HandleClientPacket {
  Packets value();

  boolean checkForRestriction() default false;
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/infrastructure/protocol/HandleClientPacket.kt
package pe.nanamochi.banchus.infrastructure.protocol

import org.springframework.stereotype.Component
import pe.nanamochi.banchus.core.PacketType

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Component
annotation class HandleClientPacket(val type: PacketType, val checkForRestriction: Boolean = false)
